### PR TITLE
Update Github workflows

### PR DESCRIPTION
**Pull request by admdly**
Update the Github workflows to remove outdated directories, improve auto-labeling, combine build and test checks, and various other changes.

Note there are pending Checks due to the required tests set on the repository - these will need updating as some have been changed/merged in this PR.

### CODE REVIEW

**Review by BelleNottelling — APPROVED** [review]
Great improvements, thanks!

**Comment by BelleNottelling** [comment]
I've removed the required tests so this can be merged once approved. Once merged, we need to ensure we select the new tests as required once again

**Review by jaapmarcus — APPROVED** [review]

**Pull request merged**
